L/LAPACK/common.jl

```diff
@@ -306,32 +304,50 @@ function lapack_script(;lapack32::Bool=false)
        -DCMAKE_TOOLCHAIN_FILE="${CMAKE_TARGET_TOOLCHAIN}" \
        -DCMAKE_BUILD_TYPE=Release \
        -DBUILD_SHARED_LIBS=ON \
+       -DTEST_FORTRAN_COMPILER=OFF \
        -DBLAS_LIBRARIES="-L${libdir} -l${BLAS}"
 
-    make -j${nproc} all
+    make -j${nproc}
     make install
 
     if [[ -f "${libdir}/libblas.${dlext}" ]]; then
         echo "Error: libblas.${dlext} has been built, linking to libblastrampoline did not work"
         exit 1
     fi
+
+    # Rename liblapack.${dlext} into liblapack32.${dlext}
+    if [[ "${LAPACK32}" == "true" ]]; then
+        mv -v ${libdir}/liblapack.${dlext} ${libdir}/liblapack32.${dlext}
+        # If there were links that are now broken, fix 'em up
+        for l in $(find ${prefix}/lib -xtype l); do
+          if [[ $(basename $(readlink ${l})) == liblapack ]]; then
+            ln -vsf liblapack32.${dlext} ${l}
+          fi
+        done
+        PATCHELF_FLAGS=()
+        # ppc64le and aarch64 have 64 kB page sizes, don't muck up the ELF section load alignment
+        if [[ ${target} == aarch64-* || ${target} == powerpc64le-* ]]; then
+          PATCHELF_FLAGS+=(--page-size 65536)
+        fi
+        if [[ ${target} == *linux* ]] || [[ ${target} == *freebsd* ]]; then
+          patchelf ${PATCHELF_FLAGS[@]} --set-soname liblapack32.${dlext} ${libdir}/liblapack32.${dlext}
+        elif [[ ${target} == *apple* ]]; then
+          install_name_tool -id liblapack32.${dlext} ${libdir}/liblapack32.${dlext}
+        fi
+    fi
     """
 end
 
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
 platforms = expand_gfortran_versions(supported_platforms())
+
 # Building ILP64 LAPACK on aarch64 linux runs into internal compiler errors with
 # GCC ≤ 7 (=> libgfortran ≤ 4).
 filter!(p -> !(arch(p) == "aarch64" && Sys.islinux(p) && libgfortran_version(p) ≤ v"4"), platforms)
```

Comment:
When I recompiled SCALAPACK32 I found that GCC 7 has a bug on some platforms if `-cpp` is used without preprocessing macros.
@ViralBShah 
Can you add a condition to use
```FFLAGS=(-cpp -ffixed-line-length-none -DUSE_ISNAN)``` only for `LAPACK_jll` and not for `LAPACK32_jll`?

Reply:
I am keeping `-ffixed-line-length-none` for all, but moved `-cpp -DUSE_ISNAN` for LAPACK_jll only.

Reply:
Didn't help with building the failing platforms.